cluster-autoscaler/core/scale_down.go

```diff
@@ -572,6 +575,50 @@ func (sd *ScaleDown) mapNodesToStatusScaleDownNodes(nodes []*apiv1.Node, nodeGro
 	return result
 }
 
+// SoftTaintUnneededNodes manage soft taints of unneeded nodes.
+func (sd *ScaleDown) SoftTaintUnneededNodes(allNodes []*apiv1.Node) (errors []error) {
+	defer metrics.UpdateDurationFromStart(metrics.ScaleDownSoftTaintUnneeded, time.Now())
+	apiCallBudget := sd.context.AutoscalingOptions.MaxBulkSoftTaintCount
+	timeBudget := sd.context.AutoscalingOptions.MaxBulkSoftTaintTime
+	skippedNodes := 0
+	startTime := now()
+	for _, node := range allNodes {
+		if deletetaint.HasToBeDeletedTaint(node) {
+			// Do not consider nodes that are scheduled to be deleted
+			continue
+		}
+		alreadyTainted := deletetaint.HasDeletionCandidateTaint(node)
+		_, unneeded := sd.unneededNodes[node.Name]
+
+		// Check if expected taints match existing taints
+		if unneeded != alreadyTainted {
+			if apiCallBudget <= 0 || now().Sub(startTime) >= timeBudget {
+				skippedNodes++
+				continue
```

Comment:
Is there any reason not to return here? I'm not convinced logging number of skipped nodes is really worth it.

Reply:
Please look at the discussion with Ola above

Reply:
but basically: on idle we scan through all of that, so optimizations during only burst down scale seems premature

Reply:
Arguably that's when we most care :)
Seriously though, I know performance impact is negligible. I meant it makes the code more complex just so we can log what seems like a not particularly important number. But I don't feel strongly about it - if you disagree let's leave it as is.